python/ufoff.py

```diff
@@ -0,0 +1,114 @@
+        if Value_Type == float:
+            builder.PrependFloat32(vector[index])
+        if Value_Type == int:
+            builder.PrependInt32(vector[index])
+        if Value_Type == bool:
+            builder.PrependBool(vector[index])
+    return builder.EndVector()
+
+
+
+
+def GetAttrReference(builder: flatbuffers.Builder, ufo: Font) -> dict: 
+    '''
+    Creates a dictionary  that contains references for each data type that needs the buffer
+    Includes a for-loop to generate a reference for the buffer-needed fields
+    To avoid NestedError
+    '''
+    attr_dict = dict()
+    # A for-loop to create a reference to all fields which needs the buffer to be created for avoiding NestedError
+    for attr in dir(ufo.info):
+        attribute_value = getattr(ufo.info, attr) 
+
+        if attr[0] != "_" and attribute_value: # put "or True" the "attribute_value" for adding empty list or None_type
+            if type(attribute_value) == str: # How about writing a BuildAttr Func
+                attr_dict[attr] = BuildAttrString(builder, attribute_value)
+
+            if type(attribute_value) == list:
+                if type(attribute_value[0]) in [int, float]: # have to discuss type conversions
+                    attr_dict[attr] = BuildAttrVector(builder, attr, attribute_value)
+
+            # TODO other nested references
+    
+    return attr_dict
+
+def BuildUfoBuffer(builder: flatbuffers.Builder, ufo: Font, attr_dict: dict):
+    '''Building the buffer with all scalar types and a reference for non-scalar types'''
+    FontInfo.Start(builder)    
+    for attr in dir(ufo.info):
+        attribute_value = getattr(ufo.info, attr)
+
+        if attr[0] != "_" and attribute_value: # Avoiding private fields and "__" functions (Double-check!)
+            if attribute_value: # Check if the field exists
+
+                # Writing scalar fields: (Other option: just use np.isscalar()!)
+                # Seeking an alternative to inclusde IntEnum in SCALARS without using "issubclass"
+                if type(attribute_value) in SCALARS or issubclass(type(attribute_value), IntEnum): 
+                    getattr(FontInfo, AddAttr(attr))(builder, attribute_value)
+
+                # writing non-scalar fields:
+                if type(attribute_value) in NON_SCALARS: # can be emplemented just with an "else"!
+                    getattr(FontInfo, AddAttr(attr))(builder, attr_dict[attr])
+                
+                # TODO other types
+
+    flat_ufo_info = FontInfo.End(builder)
+    builder.Finish(flat_ufo_info)
+    return builder.Output()
+
+    
+
+def main():
+    '''
+    Converts UFO (From UfoLib2.Font) to flatbuffer binary
+    for JSON conversion, execute:
+    flatc --json --raw-binary -o json ../schemas/ufo/fontinfo.fbs -- ufoff.bin    
+    '''
+
+    ufo = Font.open("../OswaldFont/legacy/3.0/Roman/400/src/Oswald--400.ufo")
+    builder = flatbuffers.Builder(0)
+
+    attr_dict = GetAttrReference(builder, ufo)
```

Comment:
in python function or method names are normally all lowercase and snake_case (sometimes camelCase). the capital initial letter is reserved for classes or types names. The style of flatbuffer python interface mimics the C++ API, but there's no need to follow those odd conventions in flatfont's own python code